scene/3d/world_environment.cpp

```diff
@@ -85,6 +95,17 @@ void WorldEnvironment::_update_current_camera_attributes() {
 	get_tree()->call_group_flags(SceneTree::GROUP_CALL_DEFERRED, "_world_camera_attributes_" + itos(get_viewport()->find_world_3d()->get_scenario().get_id()), "update_configuration_warnings");
 }
 
+void WorldEnvironment::_update_current_compositor() {
+	WorldEnvironment *first = Object::cast_to<WorldEnvironment>(get_tree()->get_first_node_in_group("_world_compositor_" + itos(get_viewport()->find_world_3d()->get_scenario().get_id())));
+	if (first) {
+		get_viewport()->find_world_3d()->set_compositor(first->compositor);
+	} else {
+		get_viewport()->find_world_3d()->set_camera_attributes(Ref<Compositor>());
```

Comment:
```suggestion
		get_viewport()->find_world_3d()->set_compositor(Ref<Compositor>());
```

This is probably the cause of your problem with the compositor leaking

Reply:
I wonder why that didn't give me a compile error, I'm sending a `Ref<Compositor>` to a function expecting a `Ref<CameraAttributes>`...

Reply:
Done